### ISSUE TITLE
About the scrollbuttons in the tab bar

### ISSUE THREAD

**Issue by kmarius** [qutebrowser/qutebrowser#2746]
If the number of tabs is too high, scrollbuttons should appear at the right end of the tab bar. This is currently prevented by [tabwidget.py#L733](https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/mainwindow/tabwidget.py#L733), by setting the pixelmetric for `QStyle.PM_TabBarScrollButtonWidth` to `0`. 

Due to a bug in how the bar gets drawn in `TabBarStyle`, the spacing is wrong. In the following image, the scroll button width is set to 15, 10, 5 and 0. 

![scrollbuttons](http://i.imgur.com/w7ZXRBZ.png)

As you can see, the left button is aligned to the left. Before actually fixing this I would like to know what the desired behavior is. Some options are:
* Fix the bug, set the width to 0. Scroll buttons will never be shown. This is not a bad idea for now, since you have to open a lot of tabs for it to appear. (I have added options to set min/max tab width, so I am more affected)

* Fix the bug and leave the width > 0. One would have to think about styling the buttons right, and possibly also adding options for colors, an option to enable/disable the buttons etc. Probably the better solution in the long run. (Even though I think the buttons are more like a visual help to know that there are more tabs. You will rarely be clicking them anyway)

**Comment by The-Compiler** [comment]
Are those ever shown without your options for a minimum tab width? I've never seen them. I'd prefer them not to be there - if anything, I'd prefer some kind of different/nicer "there's more tabs" indicator.

**Comment by kmarius** [comment]
Yes, this is in the latest version from github. To reproduce, you have to open some tabs and resize the window, so they get shrinked enough. Switch to the rightmost tab from another tab. Only then they will appear, because otherwise some other tab will be drawn below the (invisible, since the width is 0,) buttons. 
I have added that to the top post now.

**Comment by kmarius** [comment]
Done that. I might try to make a better indicator some time.

**Comment by The-Compiler** [comment]
So ideally (IMHO) the tabs would always exactly fill the tab bar, without any extra space/scrolling. That doesn't always seem to be what's happening, though.

**Comment by kmarius** [comment]
Right now tabs have a minimum size so the status indicator and an icon can be drawn. Without scrolling, you would need to allow tabs to be arbitrarily narrow. This would certainly be possible i.e. by disabling the indicator and/or icon, however at some point this defeats the purpose of the tab bar.

**Comment by The-Compiler** [comment]
Hm, yeah, fair enough. People should just use `gt` (`:buffer`) or `:set tabs position left` anyways at that point 😆 I'm also okay with keeping it as-is when it overflows but adding some kind of indicator.

**Comment by kmarius** [comment]
As for indicators, there is this so called tear indicator which looks like this:
![tear indicator](https://i.imgur.com/9reXC9Y.png)
(This was taken in qtcreator, because I couldn't get qutebrowser to actually draw it)
At least in recent Qt versions, there is  `QStyle::SE_TabBarTearIndicatorLeft` and `QStyle::SE_TabBarTearIndicatorRight` which suggests that one can have those on both sides.
